File: 3. Longest Substring Without Repeating Characters/longest_substring_without_repeating_characters.py
def longest_substring (text):
	len_text = len (text)
	longest_text = ""
	
	for a in range (len_text):
		for b in range (a, (len_text) - a):
			substring = text[a:a + b]
			letter = text[a + b: a + b + 1]
			print (substring + "\t\t\t" + letter + "\t\t\t" + text[a + b])
			
			if (letter in substring):
			#	break
			#else:
				if (len(longest_text) < len(substring)):
					longest_text = substring
				break
	
	return longest_text
	
w = longest_substring("pwwkew")
print (w)
print ("\n\n\n")

File: 3. Longest Substring Without Repeating Characters/longest_substring_without_repeating_characters.js
function longestSubstring (text){
	let lenText = text.length;
	let longestText = "";
	
	for (let a = 0; a < lenText; a++){
		for (let b = a; b < (lenText - a); b++){
			let subString = text.slice(a, a + b);
			let letter = text.slice (a + b, a + b + 1);
			//console.log(subString + "\t\t\t" + letter);
			
			if (subString.includes(letter)){
				if (longestText.length < subString.length){
					longestText = subString;
				}
				break;
			}
		}
	}
	
	return (longestText);
}

let w = longestSubstring("pwwkew");
console.log(w);
console.log("\n\n\n");
